File: actionlib_pkg/scripts/doDishesActionServer.py
#! /usr/bin/env python

import roslib
roslib.load_manifest('actionlib_pkg')
import rospy
import actionlib

from actionlib_pkg.msg import doDishesAction

class DoDishesServer:
  def __init__(self):
    self.server = actionlib.SimpleActionServer('do_dishes', doDishesAction, self.execute, False)
    self.server.start()

  def execute(self, goal):
    # Do lots of awesome groundbreaking robot stuff here
    print("doDishesActionServer.py-17->execute()")
    self.server.set_succeeded()


if __name__ == '__main__':
  rospy.init_node('doDishesActionServer')
  server = DoDishesServer()
  rospy.spin()

File: actionlib_pkg/src/doDishesActionServer.cpp
#include <actionlib_pkg/doDishesAction.h>  // Note: "Action" is appended
#include <actionlib/server/simple_action_server.h>

// http://wiki.ros.org/actionlib_tutorials/Tutorials

typedef actionlib::SimpleActionServer<actionlib_pkg::doDishesAction> Server;

void execute(const actionlib_pkg::doDishesGoalConstPtr& goal, Server* as)  // Note: "Action" is not appended to DoDishes here
{
  // Do lots of awesome groundbreaking robot stuff here
  ROS_INFO("actionServer.cpp-9 -> dishwasher_id: %d", goal->dishwasher_id);
  as->setSucceeded();
}

int main(int argc, char** argv)
{
  ros::init(argc, argv, "actionServer");
  ros::NodeHandle n;
  Server server(n, "do_dishes", boost::bind(&execute, _1, &server), false);
  server.start();
  ros::spin();
  return 0;
}